src/commercetools/testing/auth.py

```diff
@@ -69,3 +72,37 @@ def token(self, request):
             self.model.add_token(token)
             response = create_commercetools_response(request, json=token)
             return response
+
+    def introspect(self, request):
+        params = parse_qs(request.body.decode("utf-8"))
+
+        client_id: typing.Optional[str] = None
```

Comment:
I'm thinking getting the client_id and client_secret should be a separate function, since the logic is completely the same in the `token` function. You don't necessarily need to change this now (I can do it before releasing).

Reply:
I agree with you, this would be a good case to refactor the client id/secret out into a separate function.